### PR TITLE
CP-32863: storage driver domain support

### PR DESCRIPTION
**Pull request by edwintorok**
There is also a XAPI part coming in a separate PR, but they do not depend on each other.
Draft, because I need to do more testing.

Explanations in individual commit messages.

### CODE REVIEW

File: xc/device.ml [hunk 1]
```diff
@@ -589,6 +593,16 @@ module Vbd_Common = struct
       "params", x.params;
     ]);
 
+    let qemu_params = x.extra_backend_keys |> List.assoc_opt "qemu-params" |> Option.value ~default:"" in
+    begin match String.split_on_char ' ' qemu_params with
+    | [physical_device_path; physical_device] ->
+      List.iter (fun (k, v) -> Hashtbl.replace back_tbl k v) ([
```
Comment (lindig) [inline xc/device.ml]: The `()` around `[]` seem superfluous to me.

File: xc/hotplug.ml [hunk 2]
```diff
@@ -229,15 +229,19 @@ let run_hotplug_script device args =
     | Vbd _ -> !Xc_resources.vbd_script
     | Vif | Tap -> !Xc_resources.vif_script
     | _ -> failwith (Printf.sprintf "don't know how to run a hotplug script for: %s" kind) in
+  let backend_prefix =
```
Comment (lindig) [inline xc/hotplug.ml]: I also like the more tabular-like style a match brings:
```
match device.backend.domid with
| 0 -> ""
| id -> Printf.sprintf "/local/domain/%d/" id
```

**Comment by edwintorok** [comment]
Changed target to feature branch.

File: xc/hotplug.ml [hunk 3]
```diff
@@ -229,15 +229,20 @@ let run_hotplug_script device args =
...
       "script=" ^ script;
       "XENBUS_TYPE=" ^ kind;
       "XENBUS_PATH=" ^ (Printf.sprintf "backend/%s/%d/%d" kind device.frontend.domid device.frontend.devid);
       "XENBUS_BASE_PATH=backend";
+      "XENBUS_PREFIX=" ^ backend_prefix;
       "INTERFACE=" ^ (Printf.sprintf "%s%d.%d" kind device.frontend.domid device.frontend.devid);
     |] ] in
   try
-    debug "Running hotplug script %s %s" script (String.concat " " args);
+    debug "Running hotplug script %s %s (%s)" script (String.concat " " args) (Array.to_list env |> String.concat " ");
```
Comment (lippirk) [inline xc/hotplug.ml]: this looks like it will log all environment variables? you probably only want to log the things you added

Comment (edwintorok) [reply xc/hotplug.ml]: I should avoid logging Unix.environment (I do need all the other XENBUS* and INTERFACE values to be able to debug the script)

**Pull request merged**